File: example/jimmer-sql/src/main/java/org/babyfish/jimmer/sql/example/cfg/JacksonConfig.java
package org.babyfish.jimmer.sql.example.cfg;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectWriter;
import org.babyfish.jimmer.jackson.ImmutableModule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JacksonConfig {

    @Bean
    public ObjectWriter prettyWriter() {
        return new ObjectMapper()
                .registerModule(new ImmutableModule())
                .writerWithDefaultPrettyPrinter();
    }
}

File: example/kotlin/jimmer-sql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/sql/cfg/JacksonConfig.kt
package org.babyfish.jimmer.example.kt.sql.cfg

import org.babyfish.jimmer.jackson.ImmutableModule
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class JacksonConfig {

    // This module is important,
    // it tells spring how to serialize/deserialize jimmer objects.
    @Bean
    fun immutableModule(): ImmutableModule =
        ImmutableModule()
}